scripts/tags_and_languages.py

```diff
@@ -0,0 +1,338 @@
+                if lc == 'xx':
+                    conn.execute(f'''
+                        UPDATE {tag_type}_tags
+                        SET lc = ?, lc_tag_name = ?, lc_tag_name_concat = ?
+                        WHERE tag = ?
+                    ''', (lc, lc_tag_name, lc_tag_name_concat, tag))
+            else:
+                # Insert the new entry
+                conn.execute(f'''
+                    INSERT INTO {tag_type}_tags (tag, lc, lc_tag_name, lc_tag_name_concat)
+                    VALUES (?, ?, ?, ?)
+                ''', (tag, lc, lc_tag_name, lc_tag_name_concat))
+
+    print(f' total count of tags for {tag_type}: {conn.execute(f"SELECT COUNT(*) FROM {tag_type}_tags").fetchone()[0]}')
+
+
+def remove_accents(input_str):
+    # Normalize the string to decompose accented characters
+    normalized_str = unicodedata.normalize('NFD', input_str)
+    # Filter out the accents
+    no_accent_str = ''.join(c for c in normalized_str if unicodedata.category(c) != 'Mn')
+    return no_accent_str
+
+
+def clean_tag(tag):
+    # Use regular expressions to replace spaces with hyphens and remove unwanted characters
+    cleaned_tag = re.sub(r'[().]', '', tag).replace(" ", "-")
+    return cleaned_tag
+
+
+def update_tags(tags_text: str, old_tag_with_lc: str, old_tag: str, new_tag: str):
+    tags_text = tags_text.lower()
+
+    # Check if new_tag contains accents
+    # if contains_accents(new_tag):
+    # If new_tag contains accents, do not remove accents from tags_text
+    tags_list = tags_text.split(",")
+    tags_list = [clean_tag(x.strip()) for x in tags_list]
+
+    updated = False
+    # first, with lc as prefix: 
+    #  "Porc, en:Porc" -> "porc, fr:porc" and not "fr:porc, en:porc"
+    for i, tag in enumerate(tags_list):
+        if tag == old_tag_with_lc:
+            tags_list[i] = new_tag
+            updated = True
+            break
+    if not updated:
+        for i, tag in enumerate(tags_list):
+            if tag == old_tag:
+                tags_list[i] = new_tag
+                updated = True
+                break
+    # sometimes, tag is not having accent whereas text is having accent
+    if not updated:
+        # If new_tag does not contain accents, remove accents from tags_text
+        tags_list = remove_accents(tags_text).split(",")
+        # tags_list = [x.strip().replace(" ", "-") for x in tags_list]
+        tags_list = [clean_tag(x.strip()) for x in tags_list]
+        for i, tag in enumerate(tags_list):
+            if tag == old_tag_with_lc:
+                tags_list[i] = new_tag
+                updated = True
+                break
+    if not updated:
+        for i, tag in enumerate(tags_list):
+            if tag == old_tag:
+                tags_list[i] = new_tag
+                updated = True
+                break
+
+
+    updated_tags_text = ", ".join(tags_list)
+
+    return updated_tags_text, updated
+
+
+time_start = datetime.now()
+
+for tag_type in tag_types:
```

Comment:
It would be better to have some fuctions we call here, to keep things simple:
* one that create the tags_to_update_df
* one that creates and populate products_to_update_{tagtype}
* one that run the modifications